### PR TITLE
Added revert handling to chainlink twap oracle

### PR DESCRIPTION
**Pull request by AnonFinDev**

### CODE REVIEW

File: contracts/oracles/ChainlinkOracle.sol [hunk 1]
```diff
@@ -123,12 +127,36 @@ contract ChainlinkOracle is IOracle {
             uint256
         )
     {
-        (uint80 round, int256 latestPrice, , uint256 latestTS, ) = aggregator.getRoundData(_round);
+        (uint80 round, int256 latestPrice, , uint256 latestTS, ) = _getRoundDataWithCheck(_round);
         while (latestPrice < 0 && round > 0) {
```
Comment (protodev-rage) [inline contracts/oracles/ChainlinkOracle.sol:131]: should it be `latestPrice < 0` or `latestPrice <= 0` since default value of int256 is 0

Comment (AnonFinDev) [reply contracts/oracles/ChainlinkOracle.sol:131]: No, we want to stop when we get latestPrice = 0. Hence "latestPrice<0" is correct

**Comment by 134dd3v** [comment]
```
(uint80 round, int256 latestPrice, , uint256 latestTS, ) = aggregator.latestRoundData();
finalPrice = uint256(latestPrice);
```

Here though we know price would not make sense to be be negative, but better we ensure its not negative. Maybe just put a revert over there.

```
require(latestPrice > 0);
```

**Review by 134dd3v — APPROVED** [review]

**Review by protodev-rage — APPROVED** [review]

**Comment by AnonFinDev** [comment]
> ```
> (uint80 round, int256 latestPrice, , uint256 latestTS, ) = aggregator.latestRoundData();
> finalPrice = uint256(latestPrice);
> ```
> 
> Here though we know price would not make sense to be be negative, but better we ensure its not negative. Maybe just put a revert over there.
> 
> ```
> require(latestPrice > 0);
> ```

We need to query previous round data in case the latest price < 0 - https://github.com/rage-trade/perpswap-contracts/blob/758f7942ef05b687fd63b24b377cdd6a557a3416/contracts/oracles/ChainlinkOracle.sol#L115 

Can move the typecast into else as below.
        if (latestPrice < 0) {
            (round, finalPrice, latestTS) = _getRoundData(round - 1);
        } else {
            finalPrice = uint256(latestPrice);
        }

**Review by 134dd3v — APPROVED** [review]
lgtm

**Pull request merged**